SwiftCompilerSources/Sources/Optimizer/InstructionSimplification/SimplifyInitEnumDataAddr.swift

```diff
@@ -49,5 +49,32 @@ extension InitEnumDataAddrInst : OnoneSimplifyable {
       context.erase(instruction: inject)
       context.erase(instruction: self)
     }
+
+    // Same as above, except we allow intructions between the init_enum_data_addr and the store:
```

Comment:
This is just to show this new code working in isolation, but I assume @eeckstein you're want me to fold it with the above? (Or remove the logic above because this one is a superset of it?)

Reply:
Yes, please.
You also need to check that no instructions between the `init_enum_data_addr` and `inject_enum_addr`, except the `store`, writes to memory (`Instruction.mayWriteToMemory`)

Reply:
Please also add tests in https://github.com/apple/swift/blob/main/test/SILOptimizer/simplify_init_enum_data_addr.sil

Reply:
All done.